sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -297,6 +308,72 @@ AZ_INLINE int32_t _get_elapsed_time(int64_t current_time_msec, int64_t start_tim
   return (int32_t)elapsed_time_msec;
 }
 
+AZ_INLINE az_result _start_reconnect_timer(az_mqtt5_connection* conn, az_event event)
+{
+  az_result ret = AZ_OK;
+  int32_t random_num = 0;
+  _az_event_pipeline* pipeline = (_az_event_pipeline*)&conn->_internal.event_pipeline;
+  _az_event_pipeline_timer* timer = (_az_event_pipeline_timer*)&conn->_internal.connection_timer;
+
+  if (az_result_failed(az_platform_get_random(&random_num)))
+  {
+    az_platform_critical_error();
+  }
+  double normalized_random_num = random_num / (double)RAND_MAX;
+  int32_t random_jitter_msec
```

Comment:
There is a potential int32_t overflow here, similar to what we are fixing in azure-iot-sdk-c.

Reply:
Fixed in new commit! Thanks for pointing that out